bsm_generator/src/bsm_generator.cpp

```diff
@@ -23,9 +23,17 @@ namespace bsm_generator
 
     void BSMGenerator::initialize()
     {
+        int bsmid;
         nh_.reset(new ros::CARMANodeHandle());
         pnh_.reset(new ros::CARMANodeHandle("~"));
         pnh_->param<double>("bsm_generation_frequency", bsm_generation_frequency_, 10.0);
+        pnh_->param<bool>("bsm_id_rotation_enabled", bsm_id_rotation_enabled_, true);
+        pnh_->param<int>("bsm_message_id", bsmid);
+        for(size_t i = 0; i < 4; ++i )
+        {
+            bsm_message_id_[i] = bsmid >> (8 * i);
```

Comment:
This looks like a guaranteed segfault as you are access element i of bsm_message_id_ before it is set. 
Since you are just doing a regular iteration you can just do on this line instead
```
bsm_message_id_.emplace_back( bsmid >> (8 * i) );
```